### ISSUE TITLE
Path filtering in 1.7.2 fails with DOS C: paths

### ISSUE THREAD

**Issue by lonniev** [mitchellh/vagrant#5213]
```bash
==> vb-tt-sync: [2015-01-19T06:57:30-08:00] FATAL: Stacktrace dumped to C:/var/chef/cache/chef-stacktrace.out
==> vb-tt-sync: 
==> vb-tt-sync: [2015-01-19T06:57:30-08:00] FATAL: RegexpError: invalid backref number/name: /^C:\tmp\vagrant-chef\6c0f26688ebda1accbe1d0ab76a99d3f\cookbooks\tasktop-sync-studio\recipes\default.rb:([\d]+)/
```

```text
C:/opscode/chef/embedded/apps/chef/lib/chef/formatters/error_inspectors/resource_failure_inspector.rb:66:in `recipe_snippet'
C:/opscode/chef/embedded/apps/chef/lib/chef/formatters/error_inspectors/resource_failure_inspector.rb:43:in `add_explanation'
C:/opscode/chef/embedded/apps/chef/lib/chef/formatters/error_mapper.rb:72:in `resource_failed'
C:/opscode/chef/embedded/apps/chef/lib/chef/formatters/base.rb:139:in `resource_failed'
C:/opscode/chef/embedded/apps/chef/lib/chef/event_dispatch/dispatcher.rb:29:in `block in resource_failed'
C:/opscode/chef/embedded/apps/chef/lib/chef/event_dispatch/dispatcher.rb:29:in `each'
C:/opscode/chef/embedded/apps/chef/lib/chef/event_dispatch/dispatcher.rb:29:in `resource_failed'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource.rb:594:in `rescue in run_action'
C:/opscode/chef/embedded/apps/chef/lib/chef/resource.rb:603:in `run_action'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:49:in `run_action'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block (2 levels) in converge'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `each'
C:/opscode/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block in converge'
```
It smells like paths fed to a regex filter need to have ':' escaped to avoid being interpreted by Ruby's regex as a named reference operator.

**Comment by sethvargo** [comment]
@lonniev could this be related to your original issue here: https://github.com/mitchellh/vagrant/issues/4974?

**Comment by lonniev** [comment]
They are related through their common use of Windows paths in a world that was optimized for unix systems.

I am retesting a script that was working in 1.6.5 with a monkey patch for Chef that solved the roles directory issue by hard-coding the mapped DOS path into the list of roles directories. Now, I have removed that patch and it appears that the paths are getting found without the patch. However, a new leg of code gets executed to handle an exception and this alternate code in `resource_failure_inspector.rb` seems to have its own issue with DOS paths.

The calling recipe throws an exception because a package installation resource isn't fully idempotent--or so I think. I'm removing the file to see if the exception goes away--but that will leave the Regex issue latent in the core chef code.

**Comment by lonniev** [comment]
@sethvargo thanks for the 1.7.2 fix for chef/vagrant paths in Windows. That seems to be effective. I can toss my monkey patch away.

When I tweak my recipe to not attempt to install already installed NSIS packages--using windows_package--I avoid the additional DOS path issue in the resource_failure_inspector.rb code.

Therefore, I am at peace but you should keep your new bug open for that file.

**Comment by lonniev** [comment]
I still need the chef patch for the DOS file paths. The issue in resource_failure_inspector.rb remains but I avoid it with improved windows_package use.

**Comment by lonniev** [comment]
I ran into this same Regex parsing exception along a different flow than the role_path one. For this second case, a shell script within the guest was unable to delete a file within the guest because I happened to have the file open in the guest in an editor. No big deal. However, the chef script raised an exception and then during conclusion of its run as it attempted to be informative about which recipe was running when the exception occurred, the same `resource_failure_inspector.rb` regex itself faulted due to the presence of unescaped colons in the pathname of the recipe file.
